### PR TITLE
Feature/7792- Modal Dialog

### PR DESCRIPTION
**Pull request by karthik-aot**
*Issue #:* /bcgov/7792

*Description of changes:*

- Creating modal dialog component
- Refactoring components
- WIP progress PR

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/RoutingSlip/CreateRoutingSlip.vue [hunk 1]
```diff
@@ -45,38 +58,73 @@
...
+      </template>
+      <template v-slot:actions>
+        <v-btn large color="primary" @click="modalDialogClose()" data-test="dialog-ok-button" class="font-weight-bold">{{ modalDialogOkText }}</v-btn>
+        <v-btn large color="primary" outlined @click="modalDialogCancel()" data-test="dialog-ok-button" class="font-weight-bold" >{{ modalDialogCancelText }}</v-btn>
+      </template>
+    </ModalDialog>
   </v-container>
 </template>
-
+color
```
Comment (shabeeb-aot) [inline src/components/RoutingSlip/CreateRoutingSlip.vue:80]: looks like unwanted

File: src/components/common/ModalDialog.vue [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    :fullscreen="fullscreenOnMobile"
+    :scrollable="isScrollable"
+    :content-class="dialogClass"
+    :max-width="maxWidth"
+    v-model="isOpen"
+    @keydown.esc="cancel">
+    <v-card>
+      <v-card-title data-test="dialog-header">
+        <slot v-if="showIcon" name="icon">
+          <v-icon large color="success">mdi-check</v-icon>
```
Comment (shabeeb-aot) [inline src/components/common/ModalDialog.vue:13]: can move this to props and default value

File: src/composables/common/useModalDialog.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { computed, ref, toRefs } from '@vue/composition-api'
+
+export function useModalDialog (props, _) {
```
Comment (shabeeb-aot) [inline src/composables/common/useModalDialog.ts:3]: not needed second argument

**Review by shabeeb-aot — APPROVED** [review]

File: src/components/RoutingSlip/CreateRoutingSlip.vue (hunk 1 above)
Comment (karthik-aot) [reply src/components/RoutingSlip/CreateRoutingSlip.vue]: missed it. yeah removed

**Pull request merged**